File: apps/api/presentation/restapi/supplier_transformer.go
package restapi

import (
	"apps/api/domain"
	"encoding/json"
	apiContract "libs/api-contract"
	"net/http"
	"strconv"

	"github.com/gorilla/mux"
)

func GetSupplierId(r *http.Request) (int64, error) {
	vars := mux.Vars(r)
	idParam := vars["supplierId"]
	id, err := strconv.ParseInt(idParam, 10, 32)
	return id, err
}

func GetSupplierRequest(r *http.Request) (apiContract.SupplierRequest, error) {
	var supplierRequest apiContract.SupplierRequest
	err := json.NewDecoder(r.Body).Decode(&supplierRequest)
	return supplierRequest, err
}

func ToApiSupplier(supplier domain.Supplier) apiContract.Supplier {
	return apiContract.Supplier{
		Id:        supplier.Id,
		Name:      supplier.Name,
		Phone:     supplier.Phone,
		Address:   supplier.Address,
		MapsLink:  supplier.MapsLink,
		DeletedAt: supplier.DeletedAt,
		CreatedAt: supplier.CreatedAt,
	}
}

func ToSupplier(supplierRequest apiContract.SupplierRequest) domain.Supplier {
	return domain.Supplier{
		Name:     supplierRequest.Name,
		Phone:    supplierRequest.Phone,
		Address:  supplierRequest.Address,
		MapsLink: supplierRequest.MapsLink,
	}
}

File: libs/ui/src/data/api/supplier.transformer.ts
// eslint-disable-next-line @nx/enforce-module-boundaries
import { Supplier as ApiSupplier } from '../../../../api-contract/src';
import { Supplier, SupplierForm } from '../../domain';

export function toSupplier(supplier: ApiSupplier): Supplier {
  return {
    id: supplier.id,
    createdAt: supplier.createdAt,
    name: supplier.name,
    address: supplier.address,
    mapsLink: supplier.mapsLink,
    phone: supplier.phone ?? '',
  };
}

export function toApiSupplier(form: SupplierForm) {
  return {
    name: form.name,
    phone: form.phone,
    address: form.address,
    mapsLink: form.mapsLink,
  };
}
